darts/ad/anomaly_model/filtering_am.py

```diff
@@ -89,7 +89,7 @@ def fit(
         # TODO: add support for covariates (see eg. Kalman Filter)
 
         raise_if_not(
-            type(allow_model_training) is bool,
+            type(allow_model_training) is bool,  # noqa: E721
```

Comment:
We are going to take advantage of this PR to actually comply with E721, would you mind replace the `type(...) is ...` with `instance(...)`?

Reply:
I had it that way, but the problem is - https://stackoverflow.com/a/37888668/4521646
```py
>>> instance(2, bool)
True
```
So many tests that check input fail.

Reply:
reported in https://github.com/astral-sh/ruff/issues/10948

Reply:
@madtoinou let's land it with this noqa and see if ruff will ignore this case in the future?

Reply:
Sounds good.

However, I think that one of the `type()` line that was changed to `isinstance()` doesn't pass the tests anymore, would you mind reverting the change?

Reply:
sure, just curious why `int` has a problem :/